src/gui/components/ChannelBoot.tsx

```diff
@@ -291,8 +289,21 @@ const terminalFlow = ({
                         reject('invalid key');
                         return;
                     }
+                    const [channelId, hostVersionHash] = input.split(':');
+                    if (!hostVersionHash) {
+                        reject('invalid key');
+                        return;
+                    }
+                    const clientVersionHash = md5(
+                        getVersionStringFromConfig(socketConfig),
+                    ).slice(0, 4);
+                    if (hostVersionHash !== clientVersionHash) {
+                        reject('client app version incompatible with host');
```

Comment:
I've made a note on https://github.com/playmint/playerchain-demo/issues/106 about updating this text